### PR TITLE
Update template files

### LINKED ISSUE #49: Track Meta tasks for GA
Copy task list from https://github.com/opensearch-project/opensearch-plugins/issues/41
Here is a meta issue for all plugins to track items which have to be taken care of between RC1 and GA for all the plugins.

- [x]  Plugin branching strategy 
- [x]  Backward compatibility testing https://github.com/opensearch-project/k-NN/issues/17
- [X]  ~~DCO checks From infra team(@peterzhuamazon ), DCO check will not be included for GA release~~
- [x]  Broken links check
- [x]  Standardize labels
- [x]  README.md, etc
- [x]  Release Tags
- [x]  Release notes
- [x]  Public Documentation
- [x]  Clean up branches
- [x]  Issue and PR templates
- [x]  CI workflows
- [ ]  Plugins dependencies from Maven -> only after [this](https://github.com/opensearch-project/opensearch-plugins/issues/45) ticket is closed.
- [x]  Test coverage reporting
- [X]  License header checks https://github.com/opensearch-project/k-NN/pull/41
- [x]  Plugin artifact names
- [x]  Branch protection rules https://github.com/opensearch-project/opensearch-plugins/issues/52 -> requires admin privilege

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Clean up templates
Adding broken links check using github actions
Update MAINTAINERS
Add ADMINS.md
Add SECURITY.md
Add RELEASING.md

### Issues Resolved
#49
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Release Labels
+
+Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v2.0.0`, as well as `backport`. Use release labels to target an issue or a PR for a given release. See [MAINTAINERS](MAINTAINERS.md#triage-open-issues) for more information on triaging issues.
+
+## Releasing
+
+The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [MAINTAINERS](MAINTAINERS.md).
+
+TODO
```
Comment (jmazanec15) [inline RELEASING.md:36]: Why is this TODO?

**Review by jmazanec15 — COMMENTED** [review]
Why is RFC_TEMPLATE all caps, while other templates are lower case?

File: RELEASING.md (hunk 1 above)
Comment (VijayanB) [reply RELEASING.md]: Removed

**Comment by VijayanB** [comment]
> Why is RFC_TEMPLATE all caps, while other templates are lower case?

Renamed other files too, to keep it consistent with https://github.com/opensearch-project/.github

**Review by jmazanec15 — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Pull request merged**